### ISSUE TITLE
No communicationo with FY6900 100MHz version

### ISSUE THREAD

**Issue by aimeiz** [rjcamatos/KKMoon-FY6900#5]
Hi.
Nice software, but as original FY6900 DDS Signal PC Software.exe does not work with 100MHz generator.
It recognizes device correctly, connect to it, but doesn't read current configuration and also doesn't set parameters.
It could be similar problem as in original software (Error 380). Frequency in model 100MHz is represented by 1 more digit than in 60MHz and lover models which probably gets into problem when software starts reading frequencies from device. To short variable.
Is any chance to get it fixed?

**Comment by rjcamatos** [comment]
I, when i started the project it was because of an error on original software when i try to connect the device, it was the primary reason.
The device you tested It its an kkmoon?
I just worked out on kkmoon fy6900 100MHz, it was saying the protocol was the same for fy6900 AliExpress brands só i out that brands on readme.
I can try adding a button to manual connection, don't think it will resolve, but i think the reason comes from automatic search the device, where i do a verification compating the string submitted by the device to i think fy6* don't remember how :/, maybe taking that verification code out and do a forced manual connection it will allow connect.
I the values i print out are all on float notation.
By the way did you disabled the "keep reading parameters"? That function clears all you type 
in parameters when its active.
Does the device shows the configured parameters?
I Don't know what can be since you stated it correctly connects.

**Comment by puyoooooo** [comment]
Hi i buy  the same device from aliexpress end original app   its working for me 
 i get a error witch this software "no more port devicces to search for!"
cant  choice the com port  
any issu for me ?

**Comment by rjcamatos** [comment]
Need recomplile, to allow port search, port search is limit to 10 first COM ports.
Try this out:

"To manually change the COM port number of a device:
Open Windows Device Manager > right click the device > Properties.
Under Port tab > Advanced.
Change COM port number."
From Google Wiki.

Give the new port name like COM1 ... 9




https://www.google.com/search?q=windows+change+com+port

**Comment by rjcamatos** [comment]
> Hi i buy the same device from aliexpress end original app its working for me i get a error witch this software "no more port devicces to search for!" cant choice the com port any issu for me ?

Try sugestion below

**Comment by rjcamatos** [comment]
> > Hi i buy the same device from aliexpress end original app its working for me i get a error witch this software "no more port devicces to search for!" cant choice the com port any issu for me ?
> 
> Try sugestion below

Need recomplile, to allow port search, port search is limit to 10 first COM ports.
Try this out:

"To manually change the COM port number of a device:
Open Windows Device Manager > right click the device > Properties.
Under Port tab > Advanced.
Change COM port number."
From Google Wiki.

Give the new port name like COM1 ... 9

https://www.google.com/search?q=windows+change+com+port

**Comment by puyoooooo** [comment]
> Need recomplile, to allow port search, port search is limit to 10 first COM ports. Try this out:
> 
> "To manually change the COM port number of a device: Open Windows Device Manager > right click the device > Properties. Under Port tab > Advanced. Change COM port number." From Google Wiki.
> 
> Give the new port name like COM1 ... 9
> 
> https://www.google.com/search?q=windows+change+com+port

You have right , all my com port from 1-10 are used /reserved , now i be able to generate wave form  en read status  from the device but i can write  exemple CH1 ON/OFF , evry time i check  the case after 3 sec this will uncked end nothning done, any idea ? thanks for you help

**Comment by rjcamatos** [comment]
> 

I, i don't know if can be this but just a try ...
At the top there is a button (if i remeber) "Keep Reading Parameters", when lighted it Stay on loop reading Parameters of device and clear out what is typed (Unclick the button) To state of unligth, do the required operations and click set Parameter, when want to read back values light "Keep Reading Parameters" hoope is this.

**Comment by rjcamatos** [comment]
> > Need recomplile, to allow port search, port search is limit to 10 first COM ports. Try this out:
> > "To manually change the COM port number of a device: Open Windows Device Manager > right click the device > Properties. Under Port tab > Advanced. Change COM port number." From Google Wiki.
> > Give the new port name like COM1 ... 9
> > https://www.google.com/search?q=windows+change+com+port
> 
> You have right , all my com port from 1-10 are used /reserved , now i be able to generate wave form en get status from the device but i cant not set, exemple CH1 ON/OFF , evry time i check the case after 3 sec this will uncked end nothning done, any idea ? thanks for you help

I, i don't know if can be this but just a try ...
At the top there is a button (if i remeber) "Keep Reading Parameters", when lighted it Stay on loop reading Parameters of device and clear out what is typed (Unclick the button) To state of unligth, do the required operations and click set Parameter, when want to read back values light "Keep Reading Parameters" hoope is this.

**Comment by rjcamatos** [comment]
> Hi. Nice software, but as original FY6900 DDS Signal PC Software.exe does not work with 100MHz generator. It recognizes device correctly, connect to it, but doesn't read current configuration and also doesn't set parameters. It could be similar problem as in original software (Error 380). Frequency in model 100MHz is represented by 1 more digit than in 60MHz and lover models which probably gets into problem when software starts reading frequencies from device. To short variable. Is any chance to get it fixed?

I, when i started the project it was because of an error on original software when i try to connect the device, it was the primary reason.
The device you tested It its an kkmoon?
I just worked out on kkmoon fy6900 100MHz, it was saying the protocol was the same for fy6900 AliExpress brands so i write out that brands on readme.
The values i print out its a direct conversion from the serial output to float, don't know what can it be.
Please just make sure the device its for example on set frequency screen, and connect it again.
Try disabled the "keep reading PARAMETERS" and set new parametters
Don't put on max frequency it works but on fields it will apear as 0Hz, since the value submited from the firmware on 100MHz is 0.

**Comment by puyoooooo** [comment]
Yes that right its working for me now  i was to slow to check the box end click to  set parameters.

Thanks verry much

**Comment by rjcamatos** [comment]
Please see WIKI for COM PORT not found

**Comment by rjcamatos** [comment]
Please see WIKI if not setting parameters
